source/common/router/rds_impl.cc

```diff
@@ -111,15 +109,23 @@ void RdsRouteConfigSubscription::onConfigUpdate(
                                      route_config_name_, route_config.name()));
   }
 
-  const uint64_t new_hash = MessageUtil::hash(route_config);
-  if (!config_info_ || new_hash != config_info_.value().last_config_hash_) {
-    config_info_ = {new_hash, version_info};
-    route_config_proto_ = route_config;
+  if (config_update_info_->onRdsUpdate(route_config, version_info)) {
     stats_.config_reload_.inc();
-    ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
-              new_hash);
-    for (auto* provider : route_config_providers_) {
-      provider->onConfigUpdate();
+
+    if (config_update_info_->routeConfiguration().has_vhds()) {
+      ENVOY_LOG(debug, "rds: vhds configuration present, starting vhds: config_name={} hash={}",
+                route_config_name_, config_update_info_->configHash());
+      vhds_subscription_ = std::make_unique<VhdsSubscription>(
+          config_update_info_, factory_context_, stat_prefix_, route_config_providers_);
+      vhds_subscription_->registerInitTargetWithInitManager(factory_context_.initManager());
```

Comment:
is there any thought around management server sends RouteConfiguration with vhds after the initManager has been initialized?